core/services/relay/evm/chain_reader.go

```diff
@@ -155,6 +160,10 @@ func (cr *chainReader) Start(ctx context.Context) error {
 
 // Close unregisters polling filters for bound contracts.
 func (cr *chainReader) Close() error {
+	if cr.StateMachine.State() == "Stopped" {
```

Comment:
Why is this necessary? Same for Start?

Reply:
Doesn't Start/StopOnce  cover this?

Reply:
If this is being added to suppress a double-close, then we need to fix that root problem instead of patching it.

Reply:
Fixed the root problem that was causing a second start attempt. [`7448064` (#14123)](https://github.com/smartcontractkit/chainlink/pull/14123/commits/7448064f87e2cb8621e6375998840a337fe8731b)

Reply:
If the root problem is fixed do we still need this condition?

Reply:
It is not present now, the snippet is outdated